### PR TITLE
feat: adds `useBookmarkToggle` hook

### PR DESCRIPTION
**Pull request by cesarenaldi**

### CODE REVIEW

File: packages/react/src/publication/useBookmarkToggle.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+};
+
+export type BookmarkOperation = Operation<void>;
+
+/**
+ * `useBookmarkToggle` hook let the user save or remove a publication from their bookmarks.
+ *
+ * You MUST be authenticated via {@link useWalletLogin} to use this hook.
+ *
+ * You can use the {@link Post.bookmarked} (or  {@link Comment.bookmarked}) property to determine
```
Comment (krzysu) [inline packages/react/src/publication/useBookmarkToggle.ts:19]: I saw while trying to generate typedoc yesterday that these links to Fragment props don't work, and typedoc complains about them

File: packages/react/src/publication/useBookmarkToggle.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+export type UseBookmarkToggleArgs = {
+  profile: ProfileOwnedByMe;
+  publication: AnyPublication;
+};
+
+export type BookmarkOperation = Operation<void>;
+
+/**
+ * `useBookmarkToggle` hook let the user save or remove a publication from their bookmarks.
```
Comment (krzysu) [inline packages/react/src/publication/useBookmarkToggle.ts:15]:
```suggestion
 * `useBookmarkToggle` hook lets the user save or remove a publication from their bookmarks.
```

File: packages/react/src/publication/adapters/useBookmarkToggleController.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import { BookmarkRequest, BookmarksGateway } from './BookmarksGateway';
+import { BookmarksPresenter } from './BookmarksPresenter';
+
+export function useBookmarkToggleController() {
+  const { apolloClient, publicationCacheManager } = useSharedDependencies();
+
+  const add = async (request: BookmarkRequest) => {
+    const presenter = new BookmarksPresenter(publicationCacheManager);
+    const gateway = new BookmarksGateway(apolloClient);
+    const reaction = new ToggleProperty(gateway, presenter);
```
Comment (krzysu) [inline packages/react/src/publication/adapters/useBookmarkToggleController.ts:13]:
```suggestion
    const bookmark = new ToggleProperty(gateway, presenter);
```

**Review by krzysu — APPROVED** [review]
super easy to review this way 😃

File: packages/react/src/publication/useBookmarkToggle.ts (hunk 1 above)
Comment (cesarenaldi) [reply packages/react/src/publication/useBookmarkToggle.ts:19]: Yeah, it's part of the syntax but struggled to make it work: https://typedoc.org/tags/link/

Or something I don't fully understand.

**Pull request merged**